pkgs/tools/video/recyclarr/default.nix

```diff
@@ -0,0 +1,93 @@
+{ lib
+, stdenv
+, fetchurl
+, makeWrapper
+, autoPatchelfHook
+, fixDarwinDylibNames
+, darwin
+, recyclarr
+, git
+, icu
+, testers
+, zlib
+,
+}:
+let
+  os =
+    if stdenv.isDarwin
+    then "osx"
+    else "linux";
+  arch =
+    {
+      x86_64-linux = "x64";
+      aarch64-linux = "arm64";
+      x86_64-darwin = "x64";
+      aarch64-darwin = "arm64";
+    }."${stdenv.hostPlatform.system}"
+      or (throw "Unsupported system: ${stdenv.hostPlatform.system}");
+
+  hash =
+    {
+      x64-linux_hash = "sha256-aEcinTJlO++rTeyqGJea0TWtleH6fyooA8RhT0Qj24c=";
+      arm64-linux_hash = "sha256-9Gyk09zAGzahP8FCGxj037vaK8h++3M5R2Qqop99Gs4=";
+      x64-osx_hash = "sha256-c87eOZBz+RtbIi+dlXKKVMyPI8JqYDuiaL4xOkDRFn0=";
+      arm64-osx_hash = "sha256-zSHgLXRDB6UA7V0LFgLq9ChqB40IHIJJxRqAYyVFlB8=";
+    }."${arch}-${os}_hash";
+
+  libPath = {
+    osx = "DYLD_LIBRARY_PATH : ${lib.makeLibraryPath [darwin.ICU zlib]}";
+    linux = "LD_LIBRARY_PATH : ${lib.makeLibraryPath [icu zlib]}";
+  }."${os}";
+
+in
+stdenv.mkDerivation rec {
+  pname = "recyclarr";
+  version = "5.1.0";
+
+  src = fetchurl {
+    url = "https://github.com/recyclarr/recyclarr/releases/download/v${version}/recyclarr-${os}-${arch}.tar.xz";
+    inherit hash;
+  };
+
+  sourceRoot = ".";
+
+  nativeBuildInputs = [ makeWrapper ]
+    ++ lib.optional stdenv.hostPlatform.isLinux autoPatchelfHook
+    ++ lib.optional stdenv.hostPlatform.isDarwin fixDarwinDylibNames;
+  buildInputs = [ icu zlib ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    cp recyclarr $out/bin
```

Comment:
does it work if you symlink the binary like for example here: https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/tasks/filesystems/btrfs.nix#L70

Reply:
Can't get symlinking to work here - it builds but the target of the symlink (ie `src`) is no longer accessible after the build finishes. Keeping the `cp` command here so that the binary is copied into `$out` and accessible after build finishes/ is cleaned up. 

The usage here is modeled after some of the [trivial examples on nixpkgs manual](https://github.com/NixOS/nixpkgs/blob/e7a84def4794fb783ec907440541697e7b9c7945/pkgs/development/web/bun/default.nix#L47) as well as more complicated packages such as [radarr](https://github.com/NixOS/nixpkgs/blob/da0a385a5e720b8dfb422e4f6a1c466b001e4089/pkgs/servers/radarr/default.nix) which is a similar program to recyclarr (ie a C# program distributed as a binary from a GitHub release)

Reply:
I guess copy is fine then.